### DISCUSSION TITLE
Terrible input stutters when using laptop trackpad

### DISCUSSION THREAD

**Discussion by OwenCMYK** [ppy/osu#27215] [Q&A]
### Type

Performance

### Bug description

As shown in the video, there are huge stutters when I'm using my laptop trackpad. These are only input stutters and don't affect graphical framerate at all. There's some graphical choppiness on the video but that's only in the recording, not in actual gameplay. Here are some extra things to note about this bug:
- This behavior doesn't happen on Stable, or any other programs for that matter
- The stutters don't happen unless I'm pressing buttons. So my cursor remains smooth while watching autoplay, or if I turn on NF and don't click anything
- They get worse if I have stuff open/happening in the background on my laptop. This is to be expected of a performance related issue, but it seems odd to me that the input is lagging and not the graphics
- My keyboard and trackpad drivers are up to date
- This affects all modes

Thank you for your time. This is very reproducible so let me know if you have any other questions, or tests you'd like me to perform.

### Screenshots or videos

https://github.com/ppy/osu/assets/42657486/e0200e0d-abba-4772-90bc-383bb9d957ab

### Version

2024.131.0-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/14319720/compressed-logs.zip)

**Comment by Susko3** [comment]
Try setting your frame limiter to 2x or 4x refresh rate.

Does your laptop have a windows precision trackpad? You can check in Windows Settings → Touchpad.

> The stutters don't happen unless I'm pressing buttons.

Trackpad or keyboard buttons?
If trackpad, does pressing buttons on an external mouse also cause stutters?

If you're going to record another video, also have global statistics open (Ctrl+F2) and scroll to the `Input - ` section.

**Comment by OwenCMYK** [comment]
Changing frame limiter did not fix the issue. Neither does disabling multithreading (not that you asked, but I thought I'd mention). I don't know what a precision touchpad is, but windows says I have one. I did a bunch of extensive testing and here's more stuff I found out:
- My laptop's built in keyboard buttons cause stutters
- Clicking with the touchpad buttons or external mouse buttons doesn't cause stutters
- External keyboard buttons don't cause stutters
- Using external mouse VS touchpad for aim doesn't matter, they're both affected (I previously thought it did, but further testing suggest it doesn't)
After this testing, it seems to show that it is in fact my laptop's keyboard that is the problem, and not the touchpad. I have had keyboard/typing related issues in other programs on certain occasions, so I'm beginning to think this might not be fully lazer's fault, though it's still worth noting I don't have this issue when I occasionally play stable
